src/pudl/transform/ferc1.py

```diff
@@ -32,9 +33,141 @@
 
 logger = get_logger(__name__)
 
+TABLES_LITERAL = Literal[
+    "plants_steam_ferc1",
+    "fuel_ferc1",
+]
+
+COLUMN_RENAME: dict[TABLES_LITERAL, dict[Literal["dbf", "xbrl"], dict[str, str]]] = {
+    "plants_steam_ferc1": {
+        "dbf": {
+            "respondent_id": "utility_id_ferc1",
+            "plant_name": "plant_name_ferc1",
+            "yr_const": "construction_year",
+            "plant_kind": "plant_type",
```

Comment:
But then we're duplicating the `xbrl` and `dbf` keys, though we're probably less likely to mess something up there. 

This could also be stored as a tabular data structure, with `xbrl` and `dbf` columns, and output column names as the row index, which would be equivalent to the CSV mappings that we store on a per-year basis for the EIA spreadsheet renaming.

The more complex these structures get, the more I think it makes sense to have dedicated classes that define the structures. They can also have methods that re-arrange the outputs into whatever form we need programmatically.

E.g. in the `cleanstrings` dictionaries, it's much more compact to store a list of values with the key that they map to, but if we wanted to use `Series.map()` we'd need a 1-to-1 mapping, which could easily be generated from the more compact storage dictionary.

Reply:
we chatted about this but my suggestion here is to include the original units and then convert units separately.  this is the only effective renaming post-rename and it would feel dangerous to rename to different units outside of unit conversions. I feel tempted to rename + concurrent unit conversion but that would make the rename step more complex. i think breaking these down in *very simple* composite transform steps will make it easier to understand exactly what is happening in the transform at a high level.

I also like @bendnorman 's suggestion of the reorg of this dict!